tests/versioned/apollo-federation/federated-gateway-server-setup.js

```diff
@@ -0,0 +1,191 @@
+      magazineServer = magazineService.server
+
+      const services = [
+        { name: libraryService.name, url: libraryService.url },
+        { name: bookService.name, url: bookService.url },
+        { name: magazineService.name, url: magazineService.url }
+      ]
+
+      const plugins = [...startingPlugins, instrumentationPlugin]
+
+      const gatewayService = await loadGateway(apollo, services, plugins)
+
+      gatewayServer = gatewayService.server
+
+      t.context.helper = helper
+      t.context.serverUrl = gatewayService.url
+      t.context.libraryUrl = libraryService.url
+      t.context.bookUrl = bookService.url
+      t.context.magazineUrl = magazineService.url
+    })
+
+    t.afterEach(() => {
+      gatewayServer.stop()
+      magazineServer.stop()
+      bookServer.stop()
+      libraryServer.stop()
+
+      gatewayServer = null
+      magazineServer = null
+      bookServer = null
+      libraryServer = null
+
+      helper.unload()
+      helper = null
+
+      clearCachedModules([
+        'express',
+        'apollo-server',
+        '@apollo/gateway',
+        '@apollo/federation'
+      ])
+    })
+
+    createTests(t, WEB_FRAMEWORK)
+  })
+}
+
+async function loadGateway({ ApolloServer }, services, plugins) {
+  const name = 'Gateway'
+
+  const { ApolloGateway } = require('@apollo/gateway')
+  const gateway = new ApolloGateway({
+    serviceList: services
+  })
+
+  const server = new ApolloServer({
+    gateway,
+    subscriptions: false,
+    plugins: plugins
+  })
+
+  const { url } = await server.listen({ port: 0 })
+
+  // eslint-disable-next-line no-console
+  console.log(`${name} ready at ${url}`)
+
+  return { name, url, server }
+}
+
+async function loadLibraries({ ApolloServer, gql }, plugins) {
+  const config = federatedData.getLibraryConfiguration(gql)
+  return await loadServer(ApolloServer, config, plugins)
+}
+
+async function loadBooks({ ApolloServer, gql }, plugins) {
+  const config = federatedData.getBookConfiguration(gql)
+  return await loadServer(ApolloServer, config, plugins)
+}
+
+async function loadMagazines({ ApolloServer, gql }, plugins) {
+  const config = federatedData.getMagazineConfiguration(gql)
+  return await loadServer(ApolloServer, config, plugins)
+}
+
+async function loadServer(ApolloServer, config, plugins) {
+  const { buildFederatedSchema } = require('@apollo/federation')
+
+  const { name, typeDefs, resolvers } = config
+
+  const server = new ApolloServer({
+    schema: buildFederatedSchema([{ typeDefs, resolvers }]),
+    plugins
+  })
+
+  const { url } = await server.listen({ port: 0 })
+
+  // eslint-disable-next-line no-console
+  console.log(`${name} service ready at ${url}`)
+
+  return { name, url, server }
+}
+
+function createIgnoreTransactionPlugin(nrApi) {
+  return {
+    requestDidStart() {
+      const transactionHandle = nrApi.getTransaction()
+      transactionHandle.ignore()
+    }
+  }
+}
```

Comment:
I didn't want to overly complicate things by spinning off multiple processes, at least as of yet. Particularly since we may eventually want to validate through each hop at some point.

That does make things a little awkward, though, because even without using our plugin transactions will be created via our http and express instrumentation. There's really no way to avoid this without a separate process.

May decide to go that route at some point but for now I am ignoring transactions for the sub graph services to avoid creating the typical data and then checking for ignored on transaction finished in the tests to avoid apply assertions.